joern-cli/frontends/csharpsrc2cpg/src/main/scala/io/joern/csharpsrc2cpg/astcreation/AstCreator.scala

```diff
@@ -36,6 +37,8 @@ class AstCreator(
 
   protected var parseLevel: AstParseLevel = AstParseLevel.FULL_AST
 
+  private val anonymousTypeKeyPool = new IntervalKeyPool(first = 0, last = Long.MaxValue)
```

Comment:
Could you move this and the `nextAnonymousTypeName` under the `AstCreatorHelper` instead?